lighthouse-core/test/lib/tracehouse/trace-processor-test.js

```diff
@@ -505,36 +505,65 @@ Object {
       });
     });
 
-    describe('finds correct LCP from all frames', () => {
+    describe('finds correct metrics from all frames', () => {
       it('in a trace', () => {
         const trace = TraceProcessor.computeTraceOfTab(lcpAllFramesTrace);
         expect({
+          // Main frame
           'firstContentfulPaintEvt.ts': trace.firstContentfulPaintEvt.ts,
           'largestContentfulPaintEvt.ts': trace.largestContentfulPaintEvt.ts,
-          'mainFrameIds.frameId': trace.mainFrameIds.frameId,
-          'timeOriginEvt.ts': trace.timeOriginEvt.ts,
           'timestamps.firstContentfulPaint': trace.timestamps.firstContentfulPaint,
           'timestamps.largestContentfulPaint': trace.timestamps.largestContentfulPaint,
-          'timestamps.largestContentfulPaintAllFrames': trace.timestamps.largestContentfulPaintAllFrames, // eslint-disable-line max-len
           'timings.firstContentfulPaint': trace.timings.firstContentfulPaint,
           'timings.largestContentfulPaint': trace.timings.largestContentfulPaint,
+          // All frames
+          'firstContentfulPaintAllFramesEvt.ts': trace.firstContentfulPaintAllFramesEvt.ts,
+          'largestContentfulPaintAllFramesEvt.ts': trace.largestContentfulPaintAllFramesEvt.ts,
+          'timestamps.firstContentfulPaintAllFrames': trace.timestamps.firstContentfulPaintAllFrames, // eslint-disable-line max-len
+          'timestamps.largestContentfulPaintAllFrames': trace.timestamps.largestContentfulPaintAllFrames, // eslint-disable-line max-len
+          'timings.firstContentfulPaintAllFrames': trace.timings.firstContentfulPaintAllFrames,
           'timings.largestContentfulPaintAllFrames': trace.timings.largestContentfulPaintAllFrames,
         }).toMatchInlineSnapshot(`
           Object {
+            "firstContentfulPaintAllFramesEvt.ts": 23466705983,
             "firstContentfulPaintEvt.ts": 23466886143,
+            "largestContentfulPaintAllFramesEvt.ts": 23466705983,
             "largestContentfulPaintEvt.ts": 23466886143,
-            "mainFrameIds.frameId": "207613A6AD77B492759226780A40F6F4",
-            "timeOriginEvt.ts": 23466023130,
             "timestamps.firstContentfulPaint": 23466886143,
+            "timestamps.firstContentfulPaintAllFrames": 23466705983,
             "timestamps.largestContentfulPaint": 23466886143,
             "timestamps.largestContentfulPaintAllFrames": 23466705983,
             "timings.firstContentfulPaint": 863.013,
+            "timings.firstContentfulPaintAllFrames": 682.853,
```

Comment:
what exactly did frame-metrics.html look like would you be able to deploy it somewhere publicly too? (glitch.me or surge.sh or something?)

this doesn't feel right looking at the minified trace we have, everything seems blank until 863.

Reply:
https://rift-superficial-tang.glitch.me

Reply:
ah bummer the important moment of the screenshot was throttled out! I'll file an issue to always keep the first screenshot after FP/FCP/LCP

for posterity the child frame does indeed paint first
![image](https://user-images.githubusercontent.com/2301202/104778027-4ad71600-5742-11eb-81b5-cddb2ffebce5.png)

Reply:
https://github.com/GoogleChrome/lighthouse/issues/11961

Reply:
Yeah, if the child frame didn't paint first FCP and FCP fall frames would be the same.